### PR TITLE
ENH: Adding R-squared and Adj. R_squared to summary_col

### PR DESCRIPTION
**Pull request by gyassine**
- [ ] tests added / passed.  
- [X] code/documentation is well formatted.  
- [x] properly formatted commit message. 

I've only added both R-squared and Adj. R_squared at the end of the summary_col...  I tried my best via travis-ci, but i don't know where is the problem. Though it works in python 3.6 (in travis) and 3.7 (on my computer)

### CODE REVIEW

File: statsmodels/iolib/summary2.py [hunk 1]
```diff
@@ -395,7 +395,13 @@ def _col_params(result, float_format='%.4f', stars=True):
         res.loc[idx, res.columns[0]] = res.loc[idx, res.columns[0]] + '*'
     # Stack Coefs and Std.Errors
     res = res.iloc[:, :2]
-    res = res.stack()
+    if not isinstance(result, MixedLMResultsWrapper):
```
Comment (bashtage) [inline statsmodels/iolib/summary2.py]: Why don't you just duck-type:

``` 
if hasattr(result,'rsquared'):
    # Do stuff here with rsquared
if hasattr(result,'rsquared)adj'):
    # Do stuff here with rsquared-adj
```

This is much cleaner than making ex exemption to a single model, and having to create a dependency in this module to an unrelated one.

**Comment by gyassine** [comment]
Thanks for this suggestion @bashtage

File: statsmodels/iolib/summary2.py [hunk 2]
```diff
@@ -395,6 +395,17 @@ def _col_params(result, float_format='%.4f', stars=True):
...
     res = res.iloc[:, :2]
+    res = res.iloc[:, :2]
+    rsquared = rsquared_adj = np.nan
+    if hasattr(result, 'rsquared'):
+        rsquared = result.rsquared
+    if hasattr(result, 'rsquared_adj'):
+        rsquared_adj = result.rsquared_adj
+    r_result = pd.DataFrame({'Basic': [rsquared], 'Adj.': [rsquared_adj]},
+                            index=['R-squared'])
+    for col in r_result:
```
Comment (bashtage) [inline statsmodels/iolib/summary2.py]: You should only append if at least one of the r2 is not null.

```
if np.any(r_result):
    for col in r_results:...
```

Comment (gyassine) [reply statsmodels/iolib/summary2.py]: True...
Thanks again (it's my first pull !)

**Comment by bashtage** [comment]
You need to remove the unused import

```
statsmodels/iolib/summary2.py:4:1: F401 'statsmodels.regression.mixed_linear_model.MixedLMResultsWrapper' imported but unused
```

**Comment by gyassine** [comment]
Is everything ok now ?

**Comment by bashtage** [comment]
`test_summary_col` is failing.  Click on the links above.  You have changed the output and so the test needs to be updated.

**Comment by gyassine** [comment]
I actually don't understand my mistake... 
When I changed the output (adding the R-squared line) one platform, it gave me another error **in the same function** telling me to remove it in another platform...

In the same platform also, it is not consistent. For example, in Travis, it works for everything but python 3.7... I cannot understand why  🤕 ...

**Comment by bashtage** [comment]
Something is wrong with your code.  All of the R2 are nan in which case these shouldn’t be shown.

**Comment by bashtage** [comment]
FWIW you are failing everything on Azure, so it is a bad test.

**Comment by gyassine** [comment]
@bashtage Can you explain me why it doesn't work ?
- On azure & aapveyor, I don't understand where is the problem
- On travis, now some test ask me to put again what i took out

**Comment by bashtage** [comment]
You are failing because the models you are using don't have R2, but your string is printing them

From travis

```
 \nR-squared nan       nan      \n          nan       nan      \nX   
```

Can you run the tests locally?

**Comment by gyassine** [comment]
It works fine locally... I don't understand why it show this...

**Comment by bashtage** [comment]
I have fixed this in #5995 and will use that one.  There was an issue in old pandas where np.all didn't work correctly on a dataframe and returned a series rather than a scalar.

**Comment by bashtage** [comment]
Thanks.

**Pull request closed without merging**